### ISSUE TITLE
Accuracy mismatches the paper figure.

### ISSUE THREAD

**Issue by raeganx** [BHui97/PrivateFL#2]
First of all, thank you for your generous sharing.  I am a new FL learner and trying to experiment with your code.  However, I encountered some accuracy mismatches between my results and yours from Figure5.(f) in the paper.  I haven't used the local data transformation yet, just a FedAVG+LDP setting.

I ran the command `python FedAverage.py --data='cifar10' --nclient=100 --nclass=10 --ncpc=2 --model='resnet18' --mode='LDP' --round=100 --epsilon=4 --sr=1 --lr=5e-2 --flr=1e-1 --physical_bs=3 --E=1` twice, which is from E1_avg_cifar10.sh.  The best accuracies I could get are 0.3337 and 0.3598.  But Figure5.(f) shows I was supposed to get a 0.6?

I use Anaconda instead of miniconda and once a client finishes her training, I run `self.optim.zero_grad(set_to_none=True)` to release GPU memory.  

Should I just run the command a few times more and compute an accuracy average?  But the gap to 0.6 is a little big now. 
 Could you please give me some advice on how to reproduce your results from Figure5.(f)?  Many thanks.

**Comment by BHui97** [comment]
@Yuchen413

**Comment by Yuchen413** [comment]
I will take a look and get a response ASAP.

**Comment by Yuchen413** [comment]
@raeganx Please git clone the latest version and try it again. You may use:

python FedAverage.py --data='cifar10' --nclient=100 --nclass=10 --ncpc=2 --model='resnet18' --mode='LDP' --round=100 --epsilon=4 --sr=1 --lr=1e-1 --flr=1e-1 --physical_bs=3 --E=1

I've tested it on my end; you should be able to get the paper results with the current version. Thanks!

**Comment by raeganx** [comment]
@Yuchen413  Thanks for the response.  But I used the code downloaded about 5 days ago.
Is it because the learning rate?  Actually I first followed the parameters mentioned in the paper, which includes a 1e-1 learning rate, and the results were not good.  Then I read your script and adjusted it to 5e-2 but the accuracies still looks like this: 
![1691205289757](https://github.com/BHui97/PrivateFL/assets/125975797/bbd9b74d-b044-4d70-98ef-525ed534149e)

I will try a few more times.
